marge/interval.py

```diff
@@ -28,7 +28,7 @@ def find_weekday(string_or_day):
     if isinstance(string_or_day, str):
         return _DAY_NAMES[string_or_day.lower()]
 
-    raise ValueError('Not a week day: %r' % string_or_day)
+    raise ValueError(f'Not a week day: {string_or_day}')
```

Comment:
Does `f`-strings use `repr` or `str` representation by default? :thinking: Thinking about this because `%r` is for the `repr` representation.

Reply:
You ask hard questions :grin: I didn't know before I started reading more about it now.. 

- https://docs.python.org/3/reference/lexical_analysis.html#f-strings
- https://peps.python.org/pep-0498/
- https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

> The expressions are evaluated at runtime and then formatted using the `__format__` protocol.

Looks like that uses `__str__` :thinking: So that would mean `repr(string_or_day)` would probably be more correct here then. :+1: